expression/builtin_time.go

```diff
@@ -1852,7 +1852,66 @@ type builtinPeriodDiffSig struct {
 
 // See https://dev.mysql.com/doc/refman/5.7/en/date-and-time-functions.html#function_period-diff
 func (b *builtinPeriodDiffSig) eval(row []types.Datum) (d types.Datum, err error) {
-	return d, errFunctionNotExists.GenByArgs("PERIOD_DIFF")
+	args, err := b.evalArgs(row)
+	if err != nil {
+		return types.Datum{}, errors.Trace(err)
+	}
+	if args[0].IsNull() || args[1].IsNull() {
+		return d, nil
+	}
+	// convert input to int64
+	sc := b.ctx.GetSessionVars().StmtCtx
+	period1, err := args[0].ToInt64(sc)
+	if err != nil {
+		return d, errors.Trace(err)
+	}
+	period2, err := args[1].ToInt64(sc)
+	if err != nil {
+		return d, errors.Trace(err)
+	}
+	//input check format
+	if period1 <= 0 || period2 <= 0 || period1 > 999999 || period2 > 999999 {
+		d.SetNull()
+		return d, errors.Errorf("not in the format YYMM or YYYYMM")
+	}
+	//calculate total months from A.D. 0
+	y1 := period1 / 100
+	m1 := period1 % 100
+	y2 := period2 / 100
+	m2 := period2 % 100
+	if m1 > 12 || m2 > 12 || m1 == 0 || m2 == 0 {
+		d.SetNull()
+		return d, errors.Errorf("Month not in the right format")
+	}
+	// if YY >70 get 19YY else get 20YY  and if  YYY we got error format
+	if y1 < 70 {
+		y1 += 2000
+	} else if y1 < 100 {
+		y1 += 1900
+	} else if y1 < 1000 {
```

Comment:
Thank u very much! I didn't test it.Maybe it can be report to mysql's dev team.But on the other hand, we really have 200 DC,20 DC or 10000 DC not in the format YYYY,which we never expect to reach :disappointed: Maybe we just need to ignore it unless a historian or prophet want to use this func. So I had closed this PR.